test/bst.t

```diff
@@ -245,3 +245,187 @@ Testing Environment
+
+Test memory.swap.max cgroup usage
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit memory.swap.max=1M cat /sys/fs/cgroup/bst/memory.swap.max
+	1048576
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit memory.swap.max=1G cat /sys/fs/cgroup/bst/memory.swap.max
+	1073741824
+
+	$ ! bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit memory.swap.max=-1G cat /sys/fs/cgroup/bst/memory.swap.max 2>/dev/null
+
+Test io.weight cgroup usage
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.weight=1 cat /sys/fs/cgroup/bst/io.weight
+	default 1
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.weight="default 50" cat /sys/fs/cgroup/bst/io.weight
+	default 50
+
+	$ ! bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.weight=0 cat /sys/fs/cgroup/bst/io.weight 2>/dev/null
+
+Test io.max cgroup usage
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.max="8:0 rbps=100 wbps=500 riops=50 wiops=500" cat /sys/fs/cgroup/bst/io.max
+	8:0 rbps=100 wbps=500 riops=50 wiops=500
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.max="8:0 wbps=55 wiops=65" cat /sys/fs/cgroup/bst/io.max
+	8:0 rbps=100 wbps=55 riops=50 wiops=65
+
+	$ ! bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.max=0 cat /sys/fs/cgroup/bst/io.max 2>/dev/null
+
+Test io.latency cgroup usage
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.latency="8:0 target=40" cat /sys/fs/cgroup/bst/io.latency
+	8:0 target=40
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.latency="8:0 target=1" cat /sys/fs/cgroup/bst/io.latency
+	8:0 target=1
+
+	$ ! bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit io.latency=0 cat /sys/fs/cgroup/bst/io.latency 2>/dev/null
+
+Test pids.max cgroup usage
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit pids.max=40 cat /sys/fs/cgroup/bst/pids.max
+	40
+
+	$ bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit pids.max=19 cat /sys/fs/cgroup/bst/pids.max
+	19
+
+	$ ! bst --cgroup /sys/fs/cgroup/bst --no-cgroup-remount --climit pids.max=-1 cat /sys/fs/cgroup/bst/pids.max 2>/dev/null
+
+Test mutliple climits in conjunction
+
+	$ bst --cgroup=/sys/fs/cgroup/bst --no-cgroup-remount --climit cpu.max="5000 7000" --climit cpu.weight=100 --climit memory.max=1G --climit memory.low=1M --climit memory.max=1G --climit memory.swap.max=1G --climit io.max="8:0 rbps=100 wbps=100 riops=100 wiops=100" --climit io.latency="8:0 target=20" --climit pids.max=50 cat /sys/fs/cgroup/bst/cpu.max; cat /sys/fs/cgroup/bst/cpu.weight; cat /sys/fs/cgroup/bst/memory.min; cat /sys/fs/cgroup/bst/memory.low; cat /sys/fs/cgroup/bst/memory.high; cat /sys/fs/cgroup/bst/memory.max; cat /sys/fs/cgroup/bst/memory.swap.high; cat /sys/fs/cgroup/bst/memory.swap.max; cat /sys/fs/cgroup/bst/io.weight; cat /sys/fs/cgroup/bst/io.max; cat /sys/fs/cgroup/bst/io.latency; cat /sys/fs/cgroup/bst/pids.max
```

Comment:
how about breaking this into multiple lines:

```
$ bst --cgroup=/sys/fs/cgroup/bst \
>   --no-cgroup-remount \
>   --climit cpu.max="5000 7000" \
>   --climit cpu.weight=100 \
>   --climit memory.max=1G \
>   --climit memory.low=1M \
>   --climit memory.max=1G \
>   --climit memory.swap.max=1G \
>   --climit io.max="8:0 rbps=100 wbps=100 riops=100 wiops=100" \
>   --climit io.latency="8:0 target=20" \
>   --climit pids.max=50 \
>   cat /sys/fs/cgroup/bst/cpu.max
> cat /sys/fs/cgroup/bst/cpu.weight
> cat /sys/fs/cgroup/bst/memory.min
> cat /sys/fs/cgroup/bst/memory.low
> cat /sys/fs/cgroup/bst/memory.high
> cat /sys/fs/cgroup/bst/memory.max
> cat /sys/fs/cgroup/bst/memory.swap.high
> cat /sys/fs/cgroup/bst/memory.swap.max
> cat /sys/fs/cgroup/bst/io.weight
> cat /sys/fs/cgroup/bst/io.max
> cat /sys/fs/cgroup/bst/io.latency
> cat /sys/fs/cgroup/bst/pids.max
```

Also note that only the first cat is being executed in bst, in this current form. What you can do instead is simply pass --workdir /sys/fs/cgroup, then follow that with `cat cpu.max cpu.weight memory.min memory.low ... pids.max`.

Reply:
I had some issues using multi-line inputs in the bst test suite. Is there some special syntax needed here?

Reply:
What kind of issue? The syntax I've specified above should be working (i.e, use `$` for the start of the script, then `>` for subsequent lines)